aiohomekit/controller/ip/connection.py

```diff
@@ -477,7 +495,7 @@ async def request(
 
         # WARNING: It is vital that a Host: header is present or some devices
         # will reject the request.
-        buffer.append(f"Host: {self.host}")
+        buffer.append(f"Host: {self.connected_host or self.hosts}:{self.port}")
```

Comment:
I was wondering if this would leave to invalid host headers (which can crash some of the crappier vendors), but we shouldn't be able to get here if there is no connected host? We'd be generate a http frame that we didn't sent? So is this needed? Or can we bail out earlier?

Reply:
It shouldn't be possible to get here without a connected host actually

We can generate the host header when it connects

for ipv6 we need brackets as well anyways

Reply:
3193fdad7fc2eb99216b9d465ebc2cadee591f9b